apps/mivisionx_inference_analyzer/mivisionx_inference_analyzer.py

```diff
@@ -1,532 +1,534 @@
+        self.dim = (int(w_i),int(h_i))
+        self.outputDim = (int(n_o),int(c_o),int(h_o),int(w_o))
+
+    def __del__(self):
+        self.api.annReleaseInference(self.hdl)
+
+    def runInference(self, img, out):
+        # create input.f32 file
+        img_r = img[:,:,0]
+        img_g = img[:,:,1]
+        img_b = img[:,:,2]
+        img_t = np.concatenate((img_r, img_g, img_b), 0)    
+        # copy input f32 to inference input
+        status = self.api.annCopyToInferenceInput(self.hdl, np.ascontiguousarray(img_t, dtype=np.float32), (img.shape[0]*img.shape[1]*3*4), 0)
+        if(status):
+                print('ERROR: annCopyToInferenceInput Failed ')
+        # run inference
+        status = self.api.annRunInference(self.hdl, 1)
+        if(status):
+                print('ERROR: annRunInference Failed ')
+        # copy output f32
+        status = self.api.annCopyFromInferenceOutput(self.hdl, np.ascontiguousarray(out, dtype=np.float32), out.nbytes)
+        if(status):
+                print('ERROR: annCopyFromInferenceOutput Failed ')
+        return out
+
+    def classify(self, img):
+        # create output.f32 buffer
+        out_buf = bytearray(self.outputDim[0]*self.outputDim[1]*self.outputDim[2]*self.outputDim[3]*4)
+        out = np.frombuffer(out_buf, dtype=numpy.float32)
+        # run inference & receive output
+        output = self.runInference(img, out)
+        return output
+
+# process classification output function
+def processClassificationOutput(inputImage, modelName, modelOutput):
+    # post process output file
+    start = time.time()
+    softmaxOutput = np.float32(modelOutput)
+    topIndex = []
+    topLabels = []
+    topProb = []
+    for x in softmaxOutput.argsort()[-5:]:
+        topIndex.append(x)
+        topLabels.append(labelNames[x])
+        topProb.append(softmaxOutput[x])
+    end = time.time()
+    if(verbosePrint):
+        print('%30s' % 'Processed results in ', str((end - start)*1000), 'ms')
+
+    # display output
+    start = time.time()
+    # initialize the result image
+    resultImage = np.zeros((250, 525, 3), dtype="uint8")
+    resultImage.fill(255)
+    cv2.putText(resultImage, 'MIVisionX Object Classification', (25,  25),cv2.FONT_HERSHEY_SIMPLEX, 1.0, (0, 0, 0), 2)
+    topK = 1   
+    for i in reversed(range(5)):
+        txt =  topLabels[i]
+        conf = topProb[i]
+        txt = 'Top'+str(topK)+':'+txt+' '+str(int(round((conf*100), 0)))+'%' 
+        size = cv2.getTextSize(txt, cv2.FONT_HERSHEY_SIMPLEX, 0.5, 1)
+        t_height = size[0][1]
+        textColor = (colors[topK - 1])
+        cv2.putText(resultImage,txt,(45,t_height+(topK*30+40)),cv2.FONT_HERSHEY_SIMPLEX,0.5,textColor,1)
+        topK = topK + 1
+    end = time.time()
+    if(verbosePrint):
+        print('%30s' % 'Processed results image in ', str((end - start)*1000), 'ms')
+
+    return resultImage, topIndex, topProb
+
+# MIVisionX Classifier
+if __name__ == '__main__':

```

Comment:
@kiritigowda Not sure how to fix the codacy issue. @rrawther said we can ignore it. Can you both have a look?

Reply:
Not sure why it is complaining. But you could try putting all the code into a separate function and calling that to make it more simplified.